Neos.FluidAdaptor/Classes/Core/Widget/AjaxWidgetMiddleware.php

```diff
@@ -67,38 +66,25 @@ class AjaxWidgetComponent implements ComponentInterface
      * Check if the current request contains a widget context.
      * If so dispatch it directly, otherwise continue with the next HTTP component.
      *
-     * @param ComponentContext $componentContext
-     * @return void
-     * @throws ComponentException
+     * @param ServerRequestInterface $httpRequest
+     * @param RequestHandlerInterface $next
+     * @return ResponseInterface
      */
-    public function handle(ComponentContext $componentContext)
+    public function process(ServerRequestInterface $httpRequest, RequestHandlerInterface $next): ResponseInterface
     {
-        $httpRequest = $componentContext->getHttpRequest();
         $widgetContext = $this->extractWidgetContext($httpRequest);
         if ($widgetContext === null) {
-            return;
+            return $next->handle($httpRequest);
         }
 
         $actionRequest = $this->actionRequestFactory->createActionRequest($httpRequest, ['__widgetContext' => $widgetContext]);
         $actionRequest->setControllerObjectName($widgetContext->getControllerObjectName());
-        $this->securityContext->setRequest($actionRequest);
```

Comment:
We will do this in dispatcher with the coming dispatcher-middleware change. Don't worry.